### DISCUSSION TITLE
How to capture client IP address

### DISCUSSION THREAD

**Discussion by slloyd88** [PostgREST/postgrest#3305] [Q&A] [answered]
The documentation show code that is setting the client_addr from the request.headers x-forwarded-for value. But request.headers does not have a x-forwarded-for set for me.  Is there ANY other way to capture the client's IP address?  I do see it being logged in the journalctl so I know that PostgREST is capturing it...

```
declare
  headers   json := current_setting('request.headers', true)::json;
  client_ip text := coalesce(headers->>'x-forwarded-for', '');
  accept    text := coalesce(headers->>'accept', '');
```

**Comment by laurenceisla** [comment, marked as answer]
PostgREST only shows the client's IP it in the logs, as you have mentioned. In order for you to handle the IP, your reverse proxy needs to explicitly set the [`x-forwarded-for`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) header.

For instance, if you use Nginx as a reverse proxy, you can follow the [example in the docs](https://postgrest.org/en/latest/explanations/nginx.html#nginx) and add the header inside the `location`:

```nginx
    location /api/ {
      default_type  application/json;
      proxy_hide_header Content-Location;
      add_header Content-Location  /api/$upstream_http_content_location;
      proxy_set_header  Connection "";
      proxy_http_version 1.1;
      proxy_pass http://postgrest/;
      # setting the header:
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }
```

Only then you can access the IP by selecting that header in SQL:

```sql
select current_setting('request.headers', true)::json->>'x-forwarded-for';
```
